src/services/inplayer.account.service.ts

```diff
@@ -328,3 +333,75 @@ function parseJson(value: string, fallback = {}) {
 export const canUpdateEmail = false;
 
 export const canChangePasswordWithOldPassword = true;
+
+export const initCustomerExtras = async (): Promise<ExternalData> => {
+  const [favoritesData, historyData] = await Promise.all([InPlayer.Account.getFavorites(), await InPlayer.Account.getWatchHistory({})]);
+
+  const favorites = favoritesData.data?.collection?.map((favorite: FavoritesData) => {
+    return processFavorite(favorite);
+  });
+
+  const history = historyData.data?.collection?.map((history: WatchlistHistory) => {
+    return processHistoryItem(history);
+  });
+
+  return {
+    favorites,
+    history,
+  };
+};
+
+export const updatePersonalShelves: UpdatePersonalShelves = async (payload) => {
```

Comment:
We were talking about a downside to the current cleeng implementation today being that it updates the whole list object instead of doing operations on individual items.  This causes problems when updating from 2 different devices that have different lists cached.  The last write overwrites any intermediate changes from the other device.

Since the Inplayer API supports individual updates, it would be nice to call add/remove/update items directly from the appropriate methods in the controllers.